apps/auxo-app/src/utils/index.ts

```diff
@@ -13,10 +14,19 @@ export const prettyNumber = (n?: number): string =>
 export const toScale = (amount: number, decimals: number) =>
   BigNumber.from(amount).mul(BigNumber.from(10).pow(decimals));
 
-export const fromScale = (n: number | BigNumber, decimals: number): number => {
-  return typeof n === "number"
-    ? n
-    : n.div(BigNumber.from(10).pow(BigNumber.from(decimals))).toNumber();
+export const fromScale = (
+  n: number | BigNumber | undefined,
+  decimals: number,
+  precision = 0
+): number => {
+  if (!n) return 0;
+  if (typeof n === "number") return n;
+
+  // ethers BN implementaion ignores decimal point conversion when dividing, so we need
+  // to use an alternative approach for decimals >= 0.5
+  const numerator = Big(n.toString());
+  const raw = numerator.div(Big(10).pow(decimals)).toFixed(precision);
```

Comment:
Looks like this is the only place where we use `big.js`, is it worth adding the extra dependency?
You could try something like this instead:

```
export function getNormalizedNumber(number, decimals) {
  return new BigNumber(number).dividedBy(new BigNumber(10).pow(decimals));
}

export const roundDownLabel = (numberString) => {
  const number = parseFloat(numberString);
  return (Math.floor(number * 100) / 100).toString();
};
let label = roundDownLabel(getNormalizedNumber(BigNumber(balance), decimal).toString())
```

Reply:
Just need to ensure label doesn't get erroneously rounded down. Will be a solution for sure